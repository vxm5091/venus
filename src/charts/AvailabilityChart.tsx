
import React, { useState, useEffect, useContext } from 'react';
import Line from'@ant-design/charts/es/line';
import { historicalContext } from '../contexts/historicalContext';     
const Availability: React.FC = () => {
  
  const { aggregate, serviceData } = useContext(historicalContext)
  const [data, setData] = useState([]);

  let config = {
    data: serviceData,
    xField: 'year',
    yField: 'value',
    seriesField: 'category',
    xAxis: { type: 'time' },
    yAxis : { 
      label: {
        formatter: function formatter(v:any) {
          return ''.concat(v).replace(/\d{1,3}(?=(\d{3})+$)/g, function (s) {
            return ''.concat(s, ',');
          });
        },
      },
    },
  };
  console.log(config.data, 'config')
  if (config.data === undefined) return <div>loading</div>
  return <Line {...config} />;
};
export { Availability } ;  
