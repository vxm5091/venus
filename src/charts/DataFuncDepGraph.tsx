// this is the kind of type of the tree node
interface IByMethod {
    GET?: IService;
    POST?: IService;
    PUT?: IService;
    PATCH?: IService;
    status?: string | number;
}

interface IService {
    service?:string;
    status?: string | number;
    load: string | number;
    response_time: number;
    error: number;
    availability: number;
    key?: number;
    byMethod?: IByMethod

}

interface IDataNode {
    services: IService[];
  }

const test: IDataNode = {
    services: [
        {
            service: 'gewgul api',
            // status: 'good',
            load: '1.00 hpm',
            response_time:1266,
            error: 50,
            availability: 100,
            byMethod: {
                GET: {
                    // status: 'good',
                    load: '0.33 hpm',
                    response_time: 1200,
                    error: 30,
                    availability:100
                },
                POST: {
                  // status: 'bad',
                  load: '0.33 hpm',
                  response_time: 1200,
                  error: 50,
                  availability:100
              },
              PATCH: {
                // status: 'fair',
                load: '0.33 hpm',
                response_time: 1200,
                error: 7,
                availability:100
            }
            }
        },
        {
            service: 'wizard api',
            // status: 'fair',
            load: '1.00 hpm',
            response_time:1266,
            error: 2,
            availability: 100,
            byMethod: {
                GET: {
                    // status: 'bad',
                    load: '0.33 hpm',
                    response_time: 1200,
                    error: 0,
                    availability:100
                },
                POST: {
                    // status: 'bad',
                    load: '0.33 hpm',
                    response_time: 1200,
                    error: 3,
                    availability:100
                },
            }
        },
        {
          service: 'Solarwinids api',
          // status: 'good',
          load: '1.00 hpm',
          response_time:1266,
          error: 90,
          availability: 100,
          byMethod: {
              GET: {
                  // status: 5,
                  load: '0.33 hpm',
                  response_time: 1200,
                  error: 10,
                  availability:100
              },
              POST: {
                  // status: 4,
                  load: '0.33 hpm',
                  response_time: 1200,
                  error: 20,
                  availability:100
              },
          }
      }
    ]
}

/* ----------- CHANGE CHILD ARRAY ---------- */

function changeChildArr(arr: IService[]): TreeNode[]{
    // make a new array with map
    // we will be putting the services inside the array
    const newArr = arr.map( (obj: any) => {
        // declare a variable for the req method objects
        const objReqs = []
        // create request types objs to be put in the req methods array for each endpoint obj
        for (let key in obj.byMethod){
            const reqType: any = {
            // declare the service type to a service key
            service: key,
            // declare the status type to the status of that req method
            // I purposely declared the obj as "any" type so I could turn the key to a string here
            // status: obj.byMethod[key].status}
            }
            // renders correct statuses for children based on errors
            reqType['status'] = (obj.byMethod[key].error < 20) ? 'good' : (obj.byMethod[key].error >= 20 && obj.byMethod[key].error <40) ? "fair" : (obj.byMethod[key].error >= 40) ? 'bad' : 'bad'
            // push to req methods array for each endpoint obj
            objReqs.push(reqType)
        }
        // renders correct statuses for inner services based on errors
        const errorStat = (obj.error < 20) ? 'good' : (obj.error >= 20 && obj.error < 40) ? "fair" : (obj.error >= 40) ? 'bad' : 'bad'
        // the new Obj for each service w/ type
        const newObj: any = {
            service: obj.service,
            status: errorStat
           }

 /* -----------THIS IS WHERE YOU WOULD CHANGE "children" to "methods" ------- */
 
         // put the requests in as a "children" key
         
        newObj.children = objReqs
        // return the object to the array
        return newObj
    })
    // return the array
    return newArr
}

// console.log('change arr', changeChildArr(test.services))
let children = changeChildArr(test.services)[0]
console.log('children', children)
// console.log(test.services);

/* ----------- CHANGE MAIN ARRAY --------- */
// this is the kind of type of the tree node
interface TreeNode {
    service: string;
    status: string;
    isExpanded?: boolean;
    children?: TreeNode[];
  }

function changeData(obj: IDataNode): TreeNode{
    let result: TreeNode = {
    // create the main node and its children using helper function
            service: "Services",
            status: "good",
            children: changeChildArr(obj.services)
          }
    return result
    }

const treeData = changeData(test)

// ------ previous dummy data ------- //

// const treeData: TreeNode = {
//     service: "CodeSmith",
//     status: "good",
//     children: [
//       {
//         service: "Google API",
//         status: "good",
//         children: [
//           { service: "GET",
//           status: "bad"},
//           { service: "POST",
//           status: "good" },
//           { service: "PUT",
//           status: "fair" },
//           { service: "DEL",
//           status: "good" },
//           ]
//       },
//       {
//         service: "Plaid API",
//         status: "good",
//         children: [
//           { service: "PUT",
//           status: "good" },
//           { service: "DEL",
//           status: "good"},
//           ]
//       },
//       {
//         service: "Solarwinds API",
//         status: "fair",
//         children: [
//           { service: "GET",
//           status: "good" }
//           ]
//       },
//       {
//         service: "Surfline API",
//           status: "fair",
//         children: [
//           { service: "GET",
//           status: "bad" },
//           { service: "POST",
//           status: "fair" },
//           { service: "PUT",
//           status: "good" },
//           { service: "DEL",
//           status: "fair" },
//           ]
//       },
//       {
//         service: "Yelp API",
//         status: "bad",
//         children: [
//           { service: "GET",
//           status: "bad" },
//           { service: "POST",
//           status: "good" },
//           { service: "PUT",
//           status: "good" },
//           { service: "DEL",
//           status: "good" },
//           ]
//       },
//     ]
//   };

  export { changeChildArr, changeData, test, treeData, IDataNode, IService, TreeNode}