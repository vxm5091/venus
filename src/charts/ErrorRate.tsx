/**
 * @name ErrorRate
 * @desc Chart that displays in historical status page rendering historical load statistics. Child component of ChartContainer.
 */

import React, { useContext } from "react";
import Line from "@ant-design/charts/es/line";
import { historicalContext } from "../contexts/historicalContext";
const ErrorRate: React.FC = () => {
  
  const { serviceData } = useContext(historicalContext);

  let config = {
            data: serviceData.error_rate,
            xField: "timestamp",
            yField: "value",
            color: ["#FF6B3B", "#626681", "#FFC100", "#9FB40F", "#76523B", "#DAD5B5", "#0E8E89", "#E19348", "#F383A2", "#247FEA"], "paletteQualitative20": ["#FF6B3B", "#626681", "#FFC100", "#9FB40F", "#76523B", "#DAD5B5", "#0E8E89", "#E19348", "#F383A2", "#247FEA", "#2BCB95", "#B1ABF4", "#1D42C2", "#1D9ED1", "#D64BC0", "#255634", "#8C8C47", "#8CDAE5", "#8E283B", "#791DC9"],
            autoAdjust: true,            
            xAxis: {
              max: 15, 
              label: {
                formatter: function formatter(v:any) {
                  return v.slice(11,16); 
                }
             }
            },
            yAxis: {
              label: {
                formatter: function formatter(v:any) {
                  return v + '%'; 
                }
             }
            },
            seriesField: "service", 
            animation: {
              appear: {
                animation: 'path-in',
                duration: 1000,
              },
            },
          };
  console.log(config.data, 'config error_rate')
  return <Line {...config} />;
}
export { ErrorRate };