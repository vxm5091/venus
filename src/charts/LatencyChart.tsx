/**
 * @name LatencyChart
 * @desc Chart that renders historical statistics for availability on the Historical Status tab. Child component of ChartContainer
 */
import React, { useContext } from 'react';
import Line from'@ant-design/charts/es/line';
import { historicalContext } from '../contexts/historicalContext';     
const Latency: React.FC = () => {
  
  const { aggregate, serviceData } = useContext(historicalContext)
  
  let config = {
    data: serviceData.response_time,
    xField: "timestamp",
    yField: 'value',
    seriesField: 'service',
    color: ["#215B77", "#1B9CD0", "#61C9FF", "#ABDFFF", "#EFF3DE", "#FFDE94", "#FFC741", "#D09C10", "#795B16"], "paletteQualitative20": ["#215B77", "#227BA2", "#1B9CD0", "#22BAED", "#61C9FF", "#8AD4FF", "#ABDFFF", "#C9E9FF", "#EFF3DE", "#FFE9B8", "#FFDE94", "#FFD470", "#FFC741", "#EDB40A", "#D09C10", "#A37B16", "#795B16"],
    xAxis: { 
      label: {
         formatter: function formatter(v:any) {
           return v.slice(11,16); 
         }
      }
     },
    yAxis : { 
      label: {
        formatter: function formatter(v:any) {
          return ''.concat(v).replace(/\d{1,3}(?=(\d{3})+$)/g, function (s) {
            return ''.concat(s, ',');
          });
        },
      },
      min: 900, 
      max: 1300
    },
    animation: {
      appear: {
        animation: 'path-in',
        duration: 1000,
      },
    },
  };
  console.log(config.data, 'config responseTime')
  return <Line {...config} />;
};

export { Latency } ;  
