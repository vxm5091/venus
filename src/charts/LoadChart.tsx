import React, { useState, useEffect, useContext } from 'react';    
import Area  from '@ant-design/charts/es/area';
import { historicalContext } from '../contexts/historicalContext';
function LoadChart(): JSX.Element {
  const { serviceData } = useContext(historicalContext) 
    
  var config ={ 
    data : serviceData ,
    xField: 'year',
    yField:'value', 
    seriesField: 'category',
  };
  return <Area {...config} autoFit={true} />;
};
export { LoadChart };