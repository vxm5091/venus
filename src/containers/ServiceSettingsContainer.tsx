/**
 * @name Dashboard
 * @desc Right-hand side of the Main Display.  Dashboard that displays services and corresponding data.
 * The parent container for Aggregate Data, Service Container, and Chart Container.
 */

import React, { useContext, } from "react";
//  import TabContainer from './TabContainer'
import { AggregateStats } from "../components/AggregateStats";
import { dynamicContext } from "../contexts/dynamicContext";
import Divider from "antd/es/divider";
import Title from "antd/es/typography/Title";
import { EditableTable2 } from '../components/EditableTable'

function SettingsContainer(): JSX.Element {
  const { services, setServices, aggregate, setAggregate } = useContext(dynamicContext);

  return (
    <div id="dashboard">
      <AggregateStats
        error={aggregate.error}
        response_time={aggregate.response_time}
        load={aggregate.load}
        availability={aggregate.availability}
      />
      <Divider>
        <Title level={3}>Service Threshold Settings</Title>
      </Divider>
      <EditableTable2 />
    </div>
  );
}

export { SettingsContainer };
