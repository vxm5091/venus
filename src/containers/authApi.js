/* eslint-disable no-undef */
/* eslint-disable no-underscore-dangle */

// import { SignIn } from './SignInContainer';


/* eslint-disable no-param-reassign */
const axios = require('axios');



// const AUTH_URL = 'http://localhost:9999';
// const AUTH_URL = 'http://ec2-3-15-29-241.us-east-2.compute.amazonaws.com';
// const AUTH_URL = 'https://ec2-3-15-29-241.us-east-2.compute.amazonaws.com';
// const AUTH_URL = 


/**
 * Object will contain functions related to auth API calls from the desktop app
 */
const authApi = {};
authApi.login = body => axios.post(`${body.serverAddress}/login`, body);
authApi.signout = () => axios.get(`${body.serverAddress}/signout`);
authApi.refreshToken = body => axios.post(`${body.serverAddress}/refresh_token`, body);

/**
 * intercept all axios requests and append customer 'x-auth-token' header if
 * accessToken is available in localStorage
 */
axios.interceptors.request.use(req => {
  const accessToken = localStorage.getItem('accessToken');
  if (accessToken) req.headers['x-auth-token'] = accessToken;
  return req;
});

/**
 * intercept all axios responses where server reject a request due to invalid token
 * in order to refresh token and try again
 */
axios.interceptors.response.use(res => res, err => {
  const reqOriginal = err.config;
  const refreshToken = localStorage.getItem('refreshToken');
  /**
   * check for (1) a valid refresh token in local storage
   * (2) a 401 status code indicating that the access token has expired
   * (3) that a token refresh hasn't already been attempted
   */
  if (refreshToken && err.response.status === 401 && !reqOriginal._retry) {
    reqOriginal._retry = true;
    return authApi.refreshToken({ refreshToken })
    // return axios.post(`${AUTH_URL}/refresh_token`, {
    //   refreshToken,
    // })
      .then(res => {
        if (res.status === 200) {
          localStorage.setItem('accessToken', res.data.accessToken);
        // FIXME delete console.log
          console.log('Access token refreshed successfully!');
          return axios(reqOriginal);
        }
      })
      .catch(error => console.error(error));
  }
  return Promise.reject(err);
});


export default authApi;
 