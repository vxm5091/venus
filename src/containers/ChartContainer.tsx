/**
 * @name ChartContainer
 * @desc Child of Main Container container that holds and displays each Chart.
 */
import React, { useContext, useEffect, useState } from "react";
import { CardDropDown } from "../components/CardDropDown";
import { historicalContext } from "../contexts/historicalContext";
import { dynamicContext } from "../contexts/dynamicContext";
import { globalContext } from "../contexts/globalContext";
import { AggregateStats } from "../components/AggregateStats";
import { Availability } from "../charts/AvailabilityChart";
import { LoadChart } from "../charts/LoadChart";
import { ErrorRate } from "../charts/ErrorRate";
import { Latency } from "../charts/LatencyChart";
import Row from "antd/es/row";
import Col from "antd/es/col";
import Card from "antd/es/card";
import { Spin } from "antd";
import Divider from "antd/es/divider";
import Title from "antd/es/typography/Title";
import Radio from "antd/es/radio";
import Button from "antd/es/button";
import axios from "axios";
function ChartContainer(): JSX.Element {
  const [bool, setBool] = useState(false);
  const {
    aggregate,
    setAggregate,
    service,
    setService,
    setServiceData,
    timeRange,
    setTimeRange,
    currentRange,
    setCurrentRange,
  } = useContext(historicalContext);
  const { serverAddress } = useContext(globalContext);
  const { serviceNames } = useContext(dynamicContext);
  const test: any = {
    status: 0,
    load: 12,
    response_time: 12,
    error: 12,
    availability: 12,
  };

  useEffect(() => {
    axios
      .get(serverAddress + ":3000/getHistorical/aggregate")
      .then(function (response) { 
        console.log(response.data)
        setServiceData(response.data.lastHour);
        setTimeRange(response.data);
        setService("aggregate");
        setAggregate(response.data.lastHour.aggregate);
      }).then(function(response){
        setBool(true);
      })
      .catch(function (error) {
        console.log(error, "error");
      });
  }, []);

  // temporal options for chart displays
  const options = [
    { label: "Last Hour", value: "lastHour" },
    { label: "Last Day", value: "lastDay" },
    { label: "Last Week", value: "lastWeek" },
    { label: "Last Month", value: "lastMonth" },
  ];
  console.log(timeRange);

  // Radio Data selection function.
  const filterData = (e: any) => {
    console.log("radio checked", e.target.value);
    setServiceData(timeRange[e.target.value]);
    setCurrentRange(e.target.value);
  };
  console.log(currentRange, "currentRange");
  
  const refreshData = () => {
    if (service === ''){

    }
    axios.get(serverAddress + ':3000/getHistorical/' + service).then(function(response){
      console.log(response.data)
      setTimeRange(response.data)
      setServiceData(response.data[currentRange])
    })
    .catch(function(error){
      console.log(error)
    })
  };

  if (!bool) {
    return (
      <div>
        <Spin
          style={{ fontSize: "25px" }}
          className="loading"
          size="large"
          tip="Loading..."
        ></Spin>
      </div>
    );
  } else {
    return (
      <div id="chartContainer">
        <AggregateStats
          error={aggregate.error}
          response_time={aggregate.response_time}
          load={aggregate.load}
          availability={aggregate.availability}
        />
        <Divider>
          <Title level={3}>Historical Status</Title>
        </Divider>
        <div className="rangeSelectorContainer">
          <CardDropDown services={serviceNames} />
          <Radio.Group
            style={{ marginLeft: "10px" }}
            optionType="button"
            onChange={filterData}
            options={options}
            defaultValue="lastHour"
          />
          <Button
            type="primary"
            style={{ marginLeft: "10px" }}
            onClick={refreshData}
          >
            Refresh Data
          </Button>
        </div>
        <Row gutter={[32, 32]}>
          <Col span={12}>
            <Title level={5}>Availability Rate (%)</Title>
            <Card bordered={true} hoverable={true} style={{ width: "700px" }}>
              <Availability />
            </Card>
          </Col>
          <Col span={12}>
            <Title level={5}>Response Time (ms)</Title>
            <Card bordered={true} hoverable={true} style={{ width: "700px" }}>
              <Latency />
            </Card>
          </Col>
        </Row>
        <Row gutter={[32, 32]}>
          <Col span={12}>
            <Title level={5}>Error Rate (%)</Title>
            <Card bordered={true} hoverable={true} style={{ width: "700px" }}>
              <ErrorRate />
            </Card>
          </Col>
          <Col span={12}>
            <Title level={5}>Load (hpm)</Title>
            <Card bordered={true} hoverable={true} style={{ width: "700px" }}>
              <LoadChart />
            </Card>
          </Col>
        </Row>
      </div>
    );
  }
}

export { ChartContainer, Availability, LoadChart, Button, AggregateStats }