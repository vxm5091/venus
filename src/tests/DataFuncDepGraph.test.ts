import React from 'react';
import { changeChildArr, changeData, IDataNode, test, IService, TreeNode} from "../charts/DataFuncDepGraph"
import { DependencyGraph,treeData, getLinkComponent, LinkControls  } from '../charts/DependencyTree'
import { SettingsContainer } from '../containers/ServiceSettingsContainer';

// describes what the test is
describe('jest testing is in progress', () => {
  // what the specific test is
    it('jest testing activated', () => {
      // what is expected
      expect(true).toBeTruthy();
    })
  }) 

describe('jest snapshot testing', () => {
  describe('dependency graph snapshot testing', () => {
    it('the dependency chart component matches the snapshot', () => {
      expect(DependencyGraph).toMatchSnapshot();
    })

    it('the getLinkComponent component matches the snapshot', () => {
        expect(getLinkComponent).toMatchSnapshot();
    })

    it('the LinkControls component matches the snapshot', () => {
      expect(LinkControls).toMatchSnapshot();
    })
  })

  describe('dummy data snapshot testing', () => {
    it('the dummy data matches the snapshot', () => {
      expect(test).toMatchSnapshot();
    })
    
    it('the dummy data matches the snapshot', () => {
      expect(treeData).toMatchSnapshot();
    })
  })
})

describe('jest data parser testing', () => {
  it('dummy data ran through parser correctly matches post parser chart data', () => {
    expect(changeData(test)).toEqual(treeData)
  })

  describe('jest data child array parser testing', () => {
    it('check if first element from result array exists', () => {
      expect(changeChildArr(test.services)[0]).toBeTruthy();
    })

    it('check if result array is an object', () => {
      expect(typeof changeChildArr(test.services)).toBe('object');
    })
 
    it('check if first element from result array is an object', () => {
      expect(typeof changeChildArr(test.services)).toBe('object');
    })

    it('check if first element from result array has "service" property', () => {
      expect(changeChildArr(test.services)[0]).toHaveProperty('service')
    })

    it('check if first element from result array has "status" property', () => {
      expect(changeChildArr(test.services)[0]).toHaveProperty('status')
    })

    it('check if first element from result array has "children" property', () => {
      expect(changeChildArr(test.services)[0]).toHaveProperty('children')
    })

    it('check if dummy data result array from child parser has length of 2', () => {
      console.log(changeChildArr(test.services))
      expect(changeChildArr(test.services)).toHaveLength(2)
    })
  })
})

/** why we are not using interface in our jest testing file
 * https://stackoverflow.com/questions/48813882/testing-typescript-interface-with-jest
 */

//--------Ground work for mock testing---------//

// describe('mock test with dummy data', () => {
//   const mockCallback = jest.fn(x => changeChildArr(x));
  // function testChildFunc(items: IService[], callback: (arr: IService[]) => TreeNode[]){
//       callback(items);
//   }
//   testChildFunc(test.services,mockCallback);
//   it('mock test for changeChildArr with dummy data', () => {
//     // The mock function is called twice
//     // expect(mockCallback.mock.calls.length).toBe(1);

//     // The first argument of the first call to the function was 0
//     expect(mockCallback.mock.calls[0][0]).toBe(changeChildArr(test.services)[0]);
//   })

// })

//----- ground work for testing endpoints with promise testing by chai-as-promised -----//
// const chai = require("chai");
// const chaiAsPromised = require("chai-as-promised");
// chai.use(chaiAsPromised);
// const expect = chai.expect

// it('Promise should pass', function() {
//   return expect(readAndWriteToDB()).to.be.fulfilled
// })

//----- ground work for enzyme testing -----//
// import { shallow } from 'enzyme';
// // import {App} from '../app'; // Introduce the corresponding React component
// describe('Link', () => {
//   it('Renders link to Google', () => {
//     const link:any = shallow(<DependencyGraph/>)
//     expect(true).toBeTruthy();
//   });
// })


