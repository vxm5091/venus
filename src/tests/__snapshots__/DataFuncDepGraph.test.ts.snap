// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest snapshot testing dependency graph snapshot testing the LinkControls component matches the snapshot 1`] = `[Function]`;

exports[`jest snapshot testing dependency graph snapshot testing the dependency chart component matches the snapshot 1`] = `[Function]`;

exports[`jest snapshot testing dependency graph snapshot testing the getLinkComponent component matches the snapshot 1`] = `[Function]`;

exports[`jest snapshot testing dummy data snapshot testing the dummy data matches the snapshot 1`] = `
Object {
  "services": Array [
    Object {
      "availability": 100,
      "byMethod": Object {
        "GET": Object {
          "availability": 100,
          "error": 0,
          "load": "0.33 hpm",
          "response_time": 1200,
          "status": "good",
        },
      },
      "error": 0,
      "load": "1.00 hpm",
      "response_time": 1266,
      "service": "gogle api",
      "status": "good",
    },
    Object {
      "availability": 100,
      "byMethod": Object {
        "GET": Object {
          "availability": 100,
          "error": 0,
          "load": "0.33 hpm",
          "response_time": 1200,
          "status": "bad",
        },
        "POST": Object {
          "availability": 100,
          "error": 0,
          "load": "0.33 hpm",
          "response_time": 1200,
          "status": "bad",
        },
      },
      "error": 0,
      "load": "1.00 hpm",
      "response_time": 1266,
      "service": "wizard api",
      "status": "fair",
    },
  ],
}
`;

exports[`jest snapshot testing dummy data snapshot testing the dummy data matches the snapshot 2`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "service": "GET",
          "status": "good",
        },
      ],
      "service": "gogle api",
      "status": "good",
    },
    Object {
      "children": Array [
        Object {
          "service": "GET",
          "status": "bad",
        },
        Object {
          "service": "POST",
          "status": "bad",
        },
      ],
      "service": "wizard api",
      "status": "fair",
    },
  ],
  "service": "Services",
  "status": "good",
}
`;
